eth_keyfile/keyfile.py

```diff
@@ -55,6 +57,22 @@ def extract_key_from_keyfile(path_or_file_obj, password):
     return private_key
 
 
+@to_dict
+def normalize_keys(keyfile_json):
+    for key, value in keyfile_json.items():
+        if is_string(key):
+            norm_key = key.lower()
+        else:
+            norm_key = key
+
+        if is_dict(value):
+            norm_value = normalize_keys(value)
+        else:
+            norm_value = value
+
+        yield norm_key, norm_value
```

Comment:
I'm not sure this is any clearer, but an alternative if you like:

```py
def normalize_keys(keyfile_dict):
    return itemmap(recursive_key_lower, keyfile_dict)
 
recursive_key_lower = combine_argument_formatters(
    apply_formatter_if(
        is_string,
        str.lower,
    ),
    apply_formatter_if(
        is_dict,
        normalize_keys
    ),
)
```

Reply:
Combining that 1) this is a crypto library so I want it to be as simple as possible and 2) that the formatter applicators are non immediately intuitive...

I'm going to opt to leave it as is.